src/EnergyPlus/Fans.cc

```diff
@@ -3100,4 +3100,29 @@ void SetFanAirLoopNumber(EnergyPlusData &state, int const FanIndex, int const Ai
     state.dataFans->Fan(FanIndex).AirLoopNum = AirLoopNum;
 }
 
+Real64 ZoneExhaustFanBalancedAirFlow(EnergyPlusData &state, int const trailInletNode) // TRANE
+{
+    // pass in state and trial fan inlet node num, and return a design flow rate for a zone exhaust fan that has this inlet node
+
+    // Obtains and Allocates fan related parameters from input file, zone exhaust fans must be hard sized.
+    if (state.dataFans->GetFanInputFlag) { // First time subroutine has been entered
+        GetFanInput(state);
+        state.dataFans->GetFanInputFlag = false;
+    }
+
+    Real64 exhaustFanOAFlowForDesign(0.0);
+
+    // loop thru fans to find zone exhaust fans on this zone
+    for (int FanNum = 1; FanNum <= state.dataFans->NumFans; ++FanNum) {
+        if (state.dataFans->Fan(FanNum).FanType_Num == FanType_ZoneExhaust) {
+            if (state.dataFans->Fan(FanNum).InletNodeNum == trailInletNode) { // found it
+                // Scale maximum flow rate by the schedule for how flows are balanced in some other way (simple infiltration, mixing)
+                Real64 minSchedValue = GetScheduleMinValue(state, state.dataFans->Fan(FanNum).BalancedFractSchedNum);
+                exhaustFanOAFlowForDesign = (1.0 - minSchedValue) * state.dataFans->Fan(FanNum).MaxAirFlowRate;
```

Comment:
Here is where the schedule gets applied for when the zone exhaust fans are intended to be balanced by zone mixing, infiltration etc.

Reply:
OK, but this may need some extra logic and/or documentation when using ZoneAirMassFlowConservation options to do the balancing. When that is active I believe the Balanced Exhaust Fraction Schedule is ignored. @Nigusse ?

Reply:
. . . ignored in the system flow balance calculations. But this new check would require that the user enter the balanced flow schedule in order to quiet this warning.